js/post/update_post.js

```diff
@@ -428,4 +453,149 @@ function fillMisc() {
+**/
+function fillMinCreditCount(minCount) {
+    'use strict';
+
+    if (minCount >= 4) {
+        $('#min_creds').html('(4.0/4.0)')
+    } else {
+        $('#min_creds').html('(' + (minCount).toFixed(1) + '/4.0)');
+    }
+}
+
+
+/**
+ * Autofills extra 200-level courses for last minor constraint.
+ * @param {number} index The textbox number we are at
+ * @param HTMLElement} min300s Array of textbox elements to fill
+**/
+function addExtraMinCourses(index, min300s) {
+    'use strict';
+
+    for (var m = 0; m < 3; m++) {
+        if (index === 3) {
+            break;      
+        } else if (getCookie(additional_min_200s[m]) === 'active') {
+            min300s[index].value = additional_min_200s[m];
+            min300s[index].readOnly = true;
+            creditCount300and400['min'] += 0.5;
+            filled_textboxes_200 += 1;
+            index += 1;
+        }
+        
+    }
+}
+
+
+/**
+ * Checks whether a POSt is completed and updates credit count colour if it is.
+**/
+function checkPostCompleted() {
+    'use strict';
+
+    if (categories_completed['spec'] === 17) {
+        $('#spec_creds').css('color', 'green');
+    } else {
+        $('#spec_creds').css('color', 'red');
+    }
+    
+    if (categories_completed['maj'] === 13) {
+        $('#maj_creds').css('color', 'green');
+    } else {
+        $('#maj_creds').css('color', 'red');
+    } 
+
+    if (categories_completed['min'] === 6) {
+        $('#min_creds').css('color', 'green');
+    } else {
+        $('#min_creds').css('color', 'red');
+    }
+}
+
 
+/**
+ * Updates the Nav Bar on the Check My Post! page
+**/
+function updateNavPost() {
+    'use strict';
+
+    var nav_post = $('#nav-links')[0].getElementsByTagName('li')[3].getElementsByTagName('a')[0];
+
+    if (getCookie('specialist') === 'active') {
+        nav_post.innerHTML = 'Check My POSt! (' + (specCount).toFixed(1) + '/12.0)';
+        setCookie('activecount', (specCount).toFixed(1));
+    } else if (getCookie('major') === 'active') {
+        nav_post.innerHTML = 'Check My POSt! (' + (majCount).toFixed(1) + '/8.0)';
+        setCookie('activecount', (majCount).toFixed(1));
+    } else if (getCookie('minor') === 'active') {
+        nav_post.innerHTML = 'Check My POSt! (' + (minCount).toFixed(1) + '/4.0)';
+        setCookie('activecount', (minCount).toFixed(1));
+    } 
+}
+
+
+/**
+ * Updates the Nav Bar on the Graph page.
+**/
+function updateNavGraph() {
+    'use strict';
+
+    var nav_graph = $('#nav-links')[0].getElementsByTagName('li')[3].getElementsByTagName('a')[0];
+
+    if (getCookie('specialist') === 'active') {
+        nav_graph.innerHTML = 'Check My POSt! (' + getCookie('activecount') + '/12.0)';
+    } else if (getCookie('major') === 'active') {
+        nav_graph.innerHTML = 'Check My POSt! (' + getCookie('activecount') + '/8.0)';
+    } else if (getCookie('minor') === 'active') {
+        nav_graph.innerHTML = 'Check My POSt! (' + getCookie('activecount') + '/4.0)';
+    } 
+}
```

Comment:
I moved all the code that filled textboxes to a new file - hopefully that should make each file not look too long.